extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/PanacheMongoEntity.java

```diff
@@ -27,4 +29,21 @@
     public String toString() {
         return this.getClass().getSimpleName() + "<" + id + ">";
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (Objects.isNull(o) || getClass() != o.getClass()) {
```

Comment:
use `object == null` instead of `Objects.isNull(object)`.

Reply:
I'm so sorry, but why? Objects.isNull(object) executes it:
`public static boolean isNull(Object obj) {
        return obj == null;
    }`

Objects.isNull(object) is more semantic.

Reply:
There is a slight performance difference and why the indirection? `Objects.isNull` exists to be used as a Predicate, filter(Objects::isNull), lamda style. You do not need it here, so why the extra method call? https://stackoverflow.com/questions/37972859/java-util-objects-isnull-vs-object-null

Reply:
And `==` is more readability than `Objects.isNull(obj)`..